### ISSUE TITLE
Output of relevant indicators

### ISSUE THREAD

**Issue by sunyulin0421** [wonjune-kang/llm-speech-summarization#9]
Hello, @wonjune-kang 

I'm sorry to bother you again, but I have two questions about indicator output for you:

1. I see in the code that the output of the perplexity index is in the log_validation_perplexity function of the [writer.py](https://github.com/wonjune-kang/llm-speech-summarization/blob/main/writer.py) file. The code records the perplexity of audio and text respectively to TensorBoard. Here I added a print statement to print out the confusion degree, but I saw that Table 1 in the paper showed the confusion degree of test-clean and test-other respectively, may I ask how I should modify the code for these two confusion degrees to print out?

2. I don't see how various indicators in CNN/DailyMail Summaries and Llama 2-Chat 7B Summaries are calculated and output in the relevant code: ROUGE-1, 2, L,METEOR and BERTScore.

In view of the above two problems, can you give me some technical guidance and help?

Thank you very much, and sincerely wish you more excellent achievements on the road of scientific research!

**Comment by sunyulin0421** [comment]
I seem to find that the confusion output in writer is the overall confusion of the verification set, not the confusion of the test set.

**Comment by sunyulin0421** [comment]
I seem to find that the perplexity output in [writer.py](https://github.com/wonjune-kang/llm-speech-summarization/blob/main/writer.py) is the overall perplexity of the verification set, not the perplexity of the test set

**Comment by wonjune-kang** [comment]
Unfortunately, this repo doesn't include the code for running the evaluations and getting the ROUGE, METEOR, and BERTScore metrics on the test splits, as those were all implemented and run in separate jupyter notebooks. You can run inference on the appropriate dataset splits and use HuggingFace's [Evaluate library](https://huggingface.co/evaluate-metric) if you're interested in computing these scores yourself.

To get perplexity for the test sets, you should be able to simply run inference on the samples in each split, get the NLLs, and compute perplexity as is done in the [validation loop](https://github.com/wonjune-kang/llm-speech-summarization/blob/main/trainer.py#L313).

**Comment by sunyulin0421** [comment]
> Unfortunately, this repo doesn't include the code for running the evaluations and getting the ROUGE, METEOR, and BERTScore metrics on the test splits, as those were all implemented and run in separate jupyter notebooks. You can run inference on the appropriate dataset splits and use HuggingFace's [Evaluate library](https://huggingface.co/evaluate-metric) if you're interested in computing these scores yourself.
> 
> To get perplexity for the test sets, you should be able to simply run inference on the samples in each split, get the NLLs, and compute perplexity as is done in the [validation loop](https://github.com/wonjune-kang/llm-speech-summarization/blob/main/trainer.py#L313).

hello, @wonjune-kang 
As a novice, my coding skills really need to be improved, and the results I get in my evaluation code are not quite the same as the results in your paper.
Can you directly provide the calculation code of the audio confusion degree of the model on the test set and the calculation code of the related indicators?
Look forward to your help and thank you very much!

**Comment by wonjune-kang** [comment]
Unfortunately, there are no plans to provide the precise evaluation code at this time, as that would require a good amount of refactoring to make things readable and usable -- apologies for this! Please note that the purpose of this repo is to allow people to use the model proposed in our paper for inference, not necessarily to reproduce all the results of our experiments.

Also, I should note that the pre-trained model weights we released aren't exactly the same as the ones used in the experiments in our paper, as I believe we released the weights from a later checkpoint. So there should be some minor variations in results with the paper even with the exact same evaluation code.
